backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/NotificationManager.java

```diff
@@ -15,18 +15,32 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.hyperledger.bpa.impl.notification;
+package org.hyperledger.bpa.impl;
 
-import lombok.AllArgsConstructor;
-import lombok.Builder;
-import lombok.Getter;
 import lombok.NoArgsConstructor;
-import org.hyperledger.aries.api.message.BasicMessage;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperledger.bpa.controller.api.WebSocketMessageBody;
+import org.hyperledger.bpa.impl.util.Converter;
+import org.hyperledger.bpa.model.ChatMessage;
+import org.hyperledger.bpa.model.Partner;
 
-@Builder
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Slf4j
 @NoArgsConstructor
-@AllArgsConstructor
-@Getter
-public class BasicMessageReceivedEvent {
-    private BasicMessage message;
+@Singleton
+public class NotificationManager {
```

Comment:
Currently not much of a manager more like a MessageConversionService?

Reply:
I was envisioning this class to grow as we removed the ApplicationEventListener and ApplicationEvent classes.  I guess this falls into the scope and naming convention... what is a manager? what is a service?  when do you use a service over a repo? etc.. Do we need services if we have repos?  Should a manager always use a service and never a repo? Kind of need established patterns or it gets blurry and inconsistent.

What I guess I am seeing is:

Aries events -> AriesEventHandler -> SpecificEventHandler -> coordinate and delegate to Service classes -> repos
User events -> Controller -> SpecificManager -> coordinates and delegates to Service classes -> repos

If that is the case, then we are using Managers in too many places and shouldn't be used by other Managers. It's the services that should be used in many different managers/eventhandlers.